File: 0455-assign-cookies/0455-assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        
        int m = g.size(), n = s.size();
        
        sort(g.begin(), g.end());
        
        sort(s.begin(), s.end());
        
        int i = 0, j = 0, count = 0;
        
        while(i < m && j < n){
            
            if(g[i] <= s[j]){
                
                ++i;
                
                ++count;
                
            }
                
            ++j;
        }
        
        return count;
        
    }
};

File: 0455-assign-cookies/0455-assign-cookies.java
import java.util.Arrays;

class Solution {
    public int findContentChildren(int[] g, int[] s) {
        int m = g.length, n = s.length;
        
        Arrays.sort(g);
        Arrays.sort(s);
        
        int i = 0, j = 0, res = 0;
        
        while(i < m && j < n){
            
            if(g[i] <= s[j]){
                ++res;
                ++i;
            }
            
            ++j;            
        }
        
        return res;
    }
}